src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -0,0 +1,148 @@
+            {
+                string[] splitEntry = entry.Split(':', StringSplitOptions.TrimEntries);
+                BaselineFileContent[splitEntry[0]] = long.Parse(splitEntry[1]);
+            }
+        }
+        else
+        {
+            Assert.False(true, $"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
+        }
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv }, skipOnNullOrWhiteSpaceEnv: true)]
+    public void CompareArtifactsToBaseline()
+    {
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        string tempTarballDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(tempTarballDir);
+
+        Utilities.ExtractTarball(Config.SdkTarballPath, tempTarballDir, OutputHelper);
+        Utilities.ExtractTarball(Config.SourceBuiltArtifactsPath, tempTarballDir, OutputHelper);
+
+        var filePathCountMap = new Dictionary<string, int>();
+        (string FilePath, long Bytes)[] tarEntries = Directory.EnumerateFiles(tempTarballDir, "*", SearchOption.AllDirectories)
+            .Where(filepath => !filepath.Contains("SourceBuildReferencePackages"))
+            .Select(filePath =>
+            {
+                string result = filePath.Substring(tempTarballDir.Length + 1);
+                result = ProcessEntryName(result, filePathCountMap);
+                return (FilePath: result, Bytes: new FileInfo(filePath).Length);
+            })
+            .OrderBy(entry => entry.FilePath)
+            .ToArray();
+
+        foreach (var entry in tarEntries)
+        {
+            if (!BaselineFileContent.TryGetValue(entry.FilePath, out long baselineBytes))
+            {
+                OutputHelper.LogWarningMessage($"{entry.FilePath} does not exist in baseline. Adding it to the baseline file");
+            }
+            else
+            {
+                CompareFileSizes(entry.FilePath, entry.Bytes, baselineBytes);
+            }
+        }
+
+        Directory.Delete(tempTarballDir, true);
+
+        try
+        {
+            string actualFilePath = Path.Combine(DotNetHelper.LogsDirectory, $"Updated_ArtifactsSizes_{Config.TargetRid}.txt");
+            File.WriteAllLines(actualFilePath, tarEntries.Select(entry => $"{entry.FilePath}: {entry.Bytes}"));
+        }
+        catch (IOException ex)
+        {
+            throw new InvalidOperationException($"An error occurred while copying the baselines file: {BaselineFilePath}", ex);
+        }
+    }
+
+    private string ProcessEntryName(string originalName, Dictionary<string, int> filePathCountMap)
+    {
+        string result = BaselineHelper.RemoveRids(originalName);
+        result = BaselineHelper.RemoveVersions(result);
+
+        string[] patterns = {@"x\.y\.z", @"x\.y(?!\.z)"};
```

Comment:
On first read, this section takes a bit of time to understand.  I think the readability of the code code be improved by one of two small changes.  First would be to refactor this into a method with a name that is more self describing - AddDifferentiatingSuffix.  The second would be a brief comment describing what problem this code is solving.  e.g.  Because version numbers are abstracted, it is possible for duplicate entries.  This code adds a numeric suffix to differentiate duplicate entries.

Reply:
Updated in [this commit](https://github.com/dotnet/installer/pull/17233/commits/f891d665737a4b2f69336eb187a21557014fd0f9). I have refactored ProcessEntryName by 1) renaming it to ProcessFilePath so that it's more descriptive, and 2) abstracting away the code for differentiating the file paths by creating a function called AddDifferenciatingSuffix.